File: sidecar/vistructum_ml/contract.py
from dataclasses import dataclass

GRID = 64
STRIDE = 24
LABELS = ("ok", "hakenkreuz")
POSITIVE = 1
FEATURE_SPEC = "fs-1"

HEIGHT_CLIP = 4
HEIGHT_CONTEXT = 31
HEIGHT_SAMPLE_STEP = 5


@dataclass(frozen=True)
class ModelKind:
    name: str
    version: str
    channels: int
    scale: tuple
    offset: tuple


MASK = ModelKind("mask", "bf-mask-1", 1, (1.0,), (0.0,))
FULLSCAN = ModelKind(
    "fullscan",
    "bf-scan-2",
    4,
    (1.0 / HEIGHT_CLIP, 1.0, 1.0 / 64.0, 1.0),
    (-1.0, 0.0, -2.0, 0.0),
)
KINDS = {kind.name: kind for kind in (MASK, FULLSCAN)}

META_VERSION = "vistructum.model_version"
META_KIND = "vistructum.kind"
META_LABELS = "vistructum.labels"
META_FEATURE_SPEC = "vistructum.feature_spec"
META_THRESHOLD = "vistructum.threshold"
META_COMMIT = "vistructum.commit"
META_CONFIG = "vistructum.config"
META_METRICS = "vistructum.metrics"
META_MIN_VOTES = "vistructum.min_votes"
META_TTA = "vistructum.tta"
META_PREFILTER = "vistructum.prefilter"
REQUIRED_META = (META_VERSION, META_KIND, META_LABELS, META_FEATURE_SPEC, META_THRESHOLD)

INPUT_NAME = "features"
OUTPUT_NAME = "scores"

File: vistructum-inference/src/main/java/de/kylekreuter/vistructum/inference/Contract.java
package de.kylekreuter.vistructum.inference;

import java.util.List;

public final class Contract {

    public static final int GRID = 64;
    public static final int STRIDE = 24;
    public static final String FEATURE_SPEC = "fs-1";
    public static final List<String> LABELS = List.of("ok", "hakenkreuz");
    public static final int POSITIVE = 1;
    public static final String INPUT_NAME = "features";
    public static final String OUTPUT_NAME = "scores";
    public static final int MAX_DETECTIONS = 20;

    static final int HEIGHT_CLIP = 4;
    static final int HEIGHT_CONTEXT = 31;
    static final int HEIGHT_SAMPLE_STEP = 5;
    static final int LUMINANCE_PAD = 128;

    static final String META_VERSION = "vistructum.model_version";
    static final String META_KIND = "vistructum.kind";
    static final String META_LABELS = "vistructum.labels";
    static final String META_FEATURE_SPEC = "vistructum.feature_spec";
    static final String META_THRESHOLD = "vistructum.threshold";
    static final String META_COMMIT = "vistructum.commit";
    static final String META_MIN_VOTES = "vistructum.min_votes";
    static final String META_TTA = "vistructum.tta";
    static final String META_PREFILTER = "vistructum.prefilter";
    static final List<String> REQUIRED_META = List.of(META_VERSION, META_KIND, META_LABELS, META_FEATURE_SPEC,
            META_THRESHOLD);

    private Contract() {
    }
}
